portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,6 +37,8 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(json);
   }
 
+  private Gson gson;
+
   private String convertToJsonUsingGson(ArrayList array) {
     Gson gson = new Gson();
     String json = gson.toJson(array);
```

Comment:
BTW, just a convention tip (at least for C++ that's the case, but I think it also applies to other languages), if a local variable ("json" here) is only used in the return statement, simply do "return gson.toJson(array)" because future readers don't need to remember one more variable.